src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -77,7 +77,7 @@ internal static unsafe void StringToUnicodeFixedArray(string str, ushort* buffer
 
         internal static unsafe string UnicodeToStringFixedArray(ushort* buffer, int length)
         {
-            int end = SpanHelpers.IndexOf(ref *(char*)buffer, '\0', length);
+            int end = SpanHelpers.IndexOfChar(ref *(char*)buffer, '\0', length);
             if (end != -1)
```

Comment:
```suggestion
            int end = new ReadOnlySpan<char>(buffer, length).IndexOf('\0');
            if (end >= 0)
```
All these are for legacy interop. They can use public API just fine.